### ISSUE TITLE
Questions about visible_mask

### ISSUE THREAD

**Issue by aoyanl** [JeffWang987/OpenOccupancy#13]
Hi, I see visible_mask in config is set False, so during training and evaluation, the voxels which is unobserved are calculated. Is my understanding correct？

**Comment by JeffWang987** [comment]
When visible_mask=Ture, we calculate the visible region (of camera/LiDAR), which is only used for visualization.
Note that the visibility code is still under construction, and we do not use it in the training/validation phase

**Comment by aoyanl** [comment]
Will the visible mask be used in the training/validation phase later？

**Comment by JeffWang987** [comment]
We might leverage the mask later. In our early-stage experiment, we found that the mIoU of visible regions is much higher than that of invisible areas. And we encourage the community to investigate the influence of the visible mask (to boost performance or do something bigger).

**Comment by qihao-plus** [comment]
errors in 
```
  File "/mnt/intel/jupyterhub/qihao/sync_projects/3_occupancy/OpenOccupancy/projects/occ_plugin/occupancy/apis/test.py", line 142, in custom_multi_gpu_test
    save_occ(result['pred_c'], result['pred_f'], data['img_metas'], out_dir, data['visible_mask'], data['gt_occ'])
KeyError: 'visible_mask'
```

**Comment by JeffWang987** [comment]
> errors in @JeffWang987
> 
> ```
> bash tools/dist_test.sh ./projects/configs/baselines/CAM-R50_img1600_128x128x10.py \
>     ./work_dirs/CAM-R50_img1600_128x128x10/best_SSC_mean_epoch_23.pth 8 \
>     --show --show-dir ./work_dirs/CAM-R50_img1600_128x128x10/show
> ```
> 
> ```
>   File "/mnt/intel/jupyterhub/qihao/sync_projects/3_occupancy/OpenOccupancy/projects/occ_plugin/occupancy/apis/test.py", line 142, in custom_multi_gpu_test
>     save_occ(result['pred_c'], result['pred_f'], data['img_metas'], out_dir, data['visible_mask'], data['gt_occ'])
> KeyError: 'visible_mask'
> ```
> 
> extra: the model `best_SSC_mean_epoch_23.pth` is trained with `CAM-R50_img1600_128x128x10.py` `visible_mask = False` in default `CAM-R50_img1600_128x128x10.py`

Yes, we haven't released the training code with the mask. For now, it's only used for visualization.

**Comment by qihao-plus** [comment]
so I have to wait your code with mask for visualization ? Thanks!

**Comment by qihao-plus** [comment]
Also, how to visualize GT voxel in `OpenOccupancy/data/nuScenes-Occupancy/v0.0-base/scene_aa45d94630be481787f20121ce89a605/occupancy/548ca66e9adc48ca88d1bcaeeafc35da.npy` Thanks!

**Comment by JeffWang987** [comment]
> Also, how to visualize GT voxel in `OpenOccupancy/data/nuScenes-Occupancy/v0.0-base/scene_aa45d94630be481787f20121ce89a605/occupancy/548ca66e9adc48ca88d1bcaeeafc35da.npy` Thanks!
> 
> shape: (136774, 4)

The Visualization code is being reframed (might  be released in the future), you can DIY yours by referring to [MonoScene](https://github.com/astra-vision/MonoScene#visualization)

**Comment by onionysy** [comment]
> > Also, how to visualize GT voxel in `OpenOccupancy/data/nuScenes-Occupancy/v0.0-base/scene_aa45d94630be481787f20121ce89a605/occupancy/548ca66e9adc48ca88d1bcaeeafc35da.npy` Thanks!
> > shape: (136774, 4)
> 
> The Visualization code is being reframed (might be released in the future), you can DIY yours by referring to [MonoScene](https://github.com/astra-vision/MonoScene#visualization)

I see that you have a Visualization of your network in the "Visualization of CONet" section of Readme, how did you visualize it? Or do you have the code, but not yet open source?

**Comment by onionysy** [comment]
> > Also, how to visualize GT voxel in `OpenOccupancy/data/nuScenes-Occupancy/v0.0-base/scene_aa45d94630be481787f20121ce89a605/occupancy/548ca66e9adc48ca88d1bcaeeafc35da.npy` Thanks!
> > shape: (136774, 4)
> 
> The Visualization code is being reframed (might be released in the future), you can DIY yours by referring to [MonoScene](https://github.com/astra-vision/MonoScene#visualization)

We found that the current MoneScene visualization process uses pkl files, but the current openocc visualization process saves npy files, which is not the same. How can npy files be visualized?

**Comment by Qinyu-Z** [comment]
We have the same question with this.

**Comment by jackwky** [comment]
> 中的错误
> 
> ```
> bash tools/dist_test.sh ./projects/configs/baselines/CAM-R50_img1600_128x128x10.py \
>     ./work_dirs/CAM-R50_img1600_128x128x10/best_SSC_mean_epoch_23.pth 8 \
>     --show --show-dir ./work_dirs/CAM-R50_img1600_128x128x10/show
> ```
> 
> ```
>   File "/mnt/intel/jupyterhub/qihao/sync_projects/3_occupancy/OpenOccupancy/projects/occ_plugin/occupancy/apis/test.py", line 142, in custom_multi_gpu_test
>     save_occ(result['pred_c'], result['pred_f'], data['img_metas'], out_dir, data['visible_mask'], data['gt_occ'])
> KeyError: 'visible_mask'
> ```
> 
> 额外：模型默认使用`best_SSC_mean_epoch_23.pth``CAM-R50_img1600_128x128x10.py``visible_mask = False``CAM-R50_img1600_128x128x10.py`

我想问一下你这个路径./work_dirs/CAM-R50_img1600_128x128x10/show 是通过什么操作生成的？是你自己训练过的吗？

**Comment by YuxiChen-GL** [comment]
So if we can visualize the result through this open-sourced code now？

**Comment by YuxiChen-GL** [comment]
> > > Also, how to visualize GT voxel in `OpenOccupancy/data/nuScenes-Occupancy/v0.0-base/scene_aa45d94630be481787f20121ce89a605/occupancy/548ca66e9adc48ca88d1bcaeeafc35da.npy` Thanks!
> > > shape: (136774, 4)
> > 
> > 
> > The Visualization code is being reframed (might be released in the future), you can DIY yours by referring to [MonoScene](https://github.com/astra-vision/MonoScene#visualization)
> 
> I see that you have a Visualization of your network in the "Visualization of CONet" section of Readme, how did you visualize it? Or do you have the code, but not yet open source?

Did you visualize it successfully？

**Comment by JeffWang987** [comment]
Sorry for the late reply. We've been busy with other projects.

The visualization code  (tools/misc/visualizer.py) is now uploaded. Feel free to modify it as you need.

**Comment by 199509liwen** [comment]
> Sorry for the late reply. We've been busy with other projects.
> 
> The visualization code (tools/misc/visualizer.py) is now uploaded. Feel free to modify it as you need.

![image](https://github.com/user-attachments/assets/cd96bbae-3a82-43ac-a07d-74e2c1314142)
As for the recently uploaded visualisation code, in addition to the dataset, occupancy labels, and visualisation saving path, what exactly is the occ_pred_root file?

**Comment by 199509liwen** [comment]
> 所以我必须等待你的代码和 mask 进行可视化？谢谢！

The visualisation code has been updated, can you solve the visualisation problem?

**Comment by 199509liwen** [comment]
> 那么，我们现在是否可以通过这些开源代码来可视化结果呢？

The visualisation code has been updated, can you solve the visualisation problem?

**Comment by 199509liwen** [comment]
> 我们对此也有同样的问题。

The visualisation code has been updated, can you solve the visualisation problem?
